File: src/core/account.rs
use chrono::NaiveDate;
use uuid::Uuid;

#[derive(Debug, Clone)]
pub struct Account {
    pub id: Uuid,
    pub name: String,
    pub url: String,
    pub notes: String,
    pub last_checked: Option<NaiveDate>,
}

impl Account {
    pub fn new(name: impl Into<String>) -> Self {
        Self {
            id: Uuid::new_v4(),
            name: name.into(),
            url: String::new(),
            notes: String::new(),
            last_checked: None,
        }
    }
}

File: lamp-mobile/core/model/src/main/java/com/lamp/mobile/core/model/Account.kt
package com.lamp.mobile.core.model

import java.time.LocalDate
import java.util.UUID

data class Account(
    val id: UUID = UUID.randomUUID(),
    val name: String,
    val url: String = "",
    val notes: String = "",
    val lastChecked: LocalDate? = null,
)
